src/main/java/com/github/redouane59/twitter/helpers/AbstractRequestHelper.java

```diff
@@ -1,49 +1,53 @@
 package com.github.redouane59.twitter.helpers;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
 import com.github.redouane59.twitter.TwitterClient;
+import com.github.redouane59.twitter.signature.TwitterCredentials;
+import com.github.scribejava.apis.TwitterApi;
+import com.github.scribejava.core.builder.ServiceBuilder;
+import com.github.scribejava.core.model.OAuth1AccessToken;
+import com.github.scribejava.core.model.OAuthRequest;
+import com.github.scribejava.core.model.Response;
+import com.github.scribejava.core.model.Verb;
+import com.github.scribejava.core.oauth.OAuth10aService;
+import com.github.scribejava.core.oauth.OAuthService;
+
 import java.io.File;
 import java.io.IOException;
 import java.util.Map;
+import java.util.Optional;
 import java.util.concurrent.TimeUnit;
+
 import lombok.Getter;
 import lombok.extern.slf4j.Slf4j;
-import okhttp3.Cache;
-import okhttp3.OkHttpClient;
 
 @Slf4j
 @Getter
 public abstract class AbstractRequestHelper {
+  public static final int DEFAULT_RETRY_AFTER_SEC = 300;
 
-  private       int          sleepTime  = 5;
-  private final OkHttpClient httpClient = new OkHttpClient.Builder()
-      .readTimeout(60, TimeUnit.SECONDS)
-      .connectTimeout(60, TimeUnit.SECONDS)
-      .build();
-
-  public void wait(String response, String url) {
-    LOGGER.info("\n" + response + "\nWaiting ... " + url); // do a wait and return this function recursively
-    try {
-      TimeUnit.MINUTES.sleep(sleepTime);
-    } catch (InterruptedException e) {
-      LOGGER.error(e.getMessage(), e);
-      Thread.currentThread().interrupt();
-    }
+  private final TwitterCredentials twitterCredentials;
+  private final OAuth10aService service;
+  
+  public AbstractRequestHelper(TwitterCredentials twitterCredentials) {
+	  this(twitterCredentials, new ServiceBuilder(twitterCredentials.getApiKey())
+			  .apiSecret(twitterCredentials.getApiSecretKey())
+			  .build(TwitterApi.instance()));
   }
-
-  public OkHttpClient getHttpClient(String url) {
```

Comment:
I thought I had added a comment but I don't see it there.
Simple question: how should I manage this with your code, because the config is dynamic and depends on the called url :)
Thanks !

Reply:
It's better to have just one Http Client. And refactoring allows to do that. For example: OkHttpClient spawns threads, pools and consumes other resources - so having dynamic client commonly is not a good idea. With updated code clients gains ability to specify required Http Client and corresponding configuration in TwitterClient constructor. If you think that dynamic characteristics are needed from request to request: please list such cases and lets think together.

Reply:
Just an example but when I implemented some projects like https://github.com/redouane59/twitter-social-graph
I made several thousands of calls to the twitter API, and the cache time I was using was not always the same.
For example, the request to have a tweet data could have a 7 days cache whereas the request to have the list of the followers of a user could have a 1 day cache. 
Now I understand that what I was doing was really dirty (because I was instantiating X httpClient depending on the called URL), but is a "clean" solution exist for this kind of need ?  
Anyway my use case is out of the scope because your implementation is much better than mine but i'm curious about how to solve it :)